File: _c++/symulacja_organizmow/Organizmy/Rosliny/Guarana.cpp
//
//  Guarana.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "Guarana.hpp"

Guarana::Guarana(Swiat &swiat):Roslina(swiat){

}
void Guarana::Kolizja(Organizm& organizm){
    std::cout << "Organizm: " << organizm.Rysowanie() << " na pozycji " << organizm.GetPolozenie() << " zjada Guarane, jego sila rosnie" << std::endl;
    organizm.ZwiekszSile(3);
    Organizm::Kolizja(organizm);
}
char Guarana::Rysowanie() const{
    return 'g';
}
Organizm* Guarana::Urodz(){
    return new Guarana(this->swiat);
}
int Guarana::GetPrawdopodobienstwo(){
    return 20;
}
bool Guarana::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<Guarana*>(&organizm))
       return true;
    else
       return false;
}

File: _java/symulacja_organizmow/src/Symulacja/Rosliny/Guarana.java
package Symulacja.Rosliny;
import Symulacja.*;

import java.awt.*;

public class Guarana extends Roslina {
    public Guarana(Swiat swiat){
        super(swiat);
    };
    public char Rysowanie(){
        return 'g';
    };
    public Organizm Urodz(){
        return new Guarana(this.swiat);
    };
    public int GetPrawdopodobienstwo(){
        return 20;
    };
    public boolean MozeBycPartnerem(Organizm organizm){
        if(organizm instanceof Guarana)
        return true;
    else
        return false;
    };
    public void Kolizja(Organizm organizm){
        swiat.zdarzenia.DodajString("Organizm: " + organizm.Rysowanie() + " na pozycji " + organizm.GetPolozenie() + "<br/>zjada Guarane, jego sila rosnie");
        organizm.ZwiekszSile(3);
        super.Kolizja(organizm);
    }
    @Override
    public Color GetColor(){
        return new Color(0,255,0);
    };
}
